### PR TITLE
Move GSelectedThreadID to DataManager

### PR DESCRIPTION
**Pull request by florian-kuebler**
Move the GSelectedThreadID global from Capture to DataManager.
Also rename some variables to match guidelines.

Test: Compile, take capture, select different cpu tracks.
Bug: http://b/163020637

### CODE REVIEW

File: OrbitGl/App.cpp [hunk 1]
```diff
@@ -595,6 +595,7 @@ void OrbitApp::StopCapture() {
 }
 
 void OrbitApp::ClearCapture() {
+  set_selected_thread_id(0);
```
Comment (dimitry-) [inline OrbitGl/App.cpp:598]: 0 is kernel process iirc. use -1 for no-porcess/thread-selected.

File: OrbitGl/DataManager.cpp [hunk 2]
```diff
@@ -60,11 +60,21 @@ void DataManager::set_visible_functions(absl::flat_hash_set<uint64_t> visible_fu
   visible_functions_ = std::move(visible_functions);
 }
 
+void DataManager::set_selected_thread_id(ThreadID thread_id) {
```
Comment (dimitry-) [inline OrbitGl/DataManager.cpp:63]: drop ThreadID? use int32_t instead? we are using it elsewhere for pid for example.

File: OrbitGl/DataManager.h [hunk 3]
```diff
@@ -5,6 +5,8 @@
 #ifndef ORBIT_GL_DATA_MANAGER_H_
 #define ORBIT_GL_DATA_MANAGER_H_
 
+#include <CallstackTypes.h>
```
Comment (dimitry-) [inline OrbitGl/DataManager.h:8]: and this include won't be needed anymore.

File: OrbitGl/TextBox.cpp [hunk 4]
```diff
@@ -69,59 +69,59 @@ float TextBox::GetScreenSize(const TextRenderer& a_TextRenderer) {
...
-  if (Capture::GSelectedThreadId != 0 && isCoreActivity && !isSameThreadIdAsSelected) {
-    isInactive = true;
+void TextBox::Draw(Batcher* batcher, TextRenderer& text_renderer, float min_x, bool visible,
+                   bool right_justify, bool is_inactive, unsigned int id, bool is_picking,
+                   bool is_highlighted) {
+  bool is_core_activity = timer_info_.type() == TimerInfo::kCoreActivity;
+  bool is_same_thread_id_as_selected =
+      is_core_activity && timer_info_.thread_id() == GOrbitApp->selected_thread_id();
+
+  if (GOrbitApp->selected_thread_id() != 0 && is_core_activity && !is_same_thread_id_as_selected) {
```
Comment (dimitry-) [inline OrbitGl/TextBox.cpp:79]: and -1 here as well.

File: OrbitGl/SchedulerTrack.cpp [hunk 5]
```diff
@@ -26,12 +27,12 @@ float SchedulerTrack::GetHeight() const {
...
 
 bool SchedulerTrack::IsTimerActive(const TimerInfo& timer_info) const {
-  bool is_same_tid_as_selected = timer_info.thread_id() == Capture::GSelectedThreadId;
+  bool is_same_tid_as_selected = timer_info.thread_id() == GOrbitApp->selected_thread_id();
   int32_t capture_process_id = Capture::capture_data_.process_id();
   bool is_same_pid_as_target =
       capture_process_id == 0 || capture_process_id == timer_info.process_id();
 
-  return is_same_tid_as_selected || (Capture::GSelectedThreadId == 0 && is_same_pid_as_target);
+  return is_same_tid_as_selected || (GOrbitApp->selected_thread_id() == 0 && is_same_pid_as_target);
```
Comment (dimitry-) [inline OrbitGl/SchedulerTrack.cpp:35]: -1

File: OrbitGl/GpuTrack.cpp [hunk 6]
```diff
@@ -55,11 +56,11 @@ GpuTrack::GpuTrack(TimeGraph* time_graph, std::shared_ptr<StringManager> string_
 }
 
 bool GpuTrack::IsTimerActive(const TimerInfo& timer_info) const {
-  bool is_same_tid_as_selected = timer_info.thread_id() == Capture::GSelectedThreadId;
+  bool is_same_tid_as_selected = timer_info.thread_id() == GOrbitApp->selected_thread_id();
   // We do not properly track the PID for GPU jobs and we still want to show
   // all jobs as active when no thread is selected, so this logic is a bit
   // different than SchedulerTrack::IsTimerActive.
-  bool no_thread_selected = Capture::GSelectedThreadId == 0;
+  bool no_thread_selected = GOrbitApp->selected_thread_id() == 0;
```
Comment (dimitry-) [inline OrbitGl/GpuTrack.cpp:63]: -1

File: OrbitGl/EventTrack.h [hunk 7]
```diff
@@ -41,7 +41,7 @@ class EventTrack : public Track {
  protected:
   TextBox m_ThreadName;
   GlCanvas* m_Canvas;
-  ThreadID m_ThreadId;
+  ThreadID thread_id_;
```
Comment (dimitry-) [inline OrbitGl/EventTrack.h:44]: int32_t

File: OrbitGl/App.cpp (hunk 1 above)
Comment (florian-kuebler) [reply OrbitGl/App.cpp]: done.

File: OrbitGl/CaptureWindow.cpp [hunk 8]
```diff
@@ -148,7 +148,7 @@ void CaptureWindow::Pick(int a_X, int a_Y) {
   PickingId pickId = PickingId::FromPixelValue(value);
 
   Capture::GSelectedTextBox = nullptr;
-  Capture::GSelectedThreadId = 0;
+  GOrbitApp->set_selected_thread_id(0);
```
Comment (florian-kuebler) [inline OrbitGl/CaptureWindow.cpp]: used -1 here

File: OrbitGl/DataManager.cpp (hunk 2 above)
Comment (florian-kuebler) [reply OrbitGl/DataManager.cpp]: done.

File: OrbitGl/DataManager.h (hunk 3 above)
Comment (florian-kuebler) [reply OrbitGl/DataManager.h]: done.

File: OrbitGl/DataManager.h [hunk 9]
```diff
@@ -37,12 +40,14 @@ class DataManager final {
...
   [[nodiscard]] const absl::flat_hash_set<uint64_t>& selected_functions() const;
   [[nodiscard]] bool IsFunctionVisible(uint64_t function_address) const;
+  [[nodiscard]] ThreadID selected_thread_id() const;
 
  private:
   const std::thread::id main_thread_id_;
   absl::flat_hash_map<int32_t, std::unique_ptr<ProcessData>> process_map_;
   absl::flat_hash_set<uint64_t> selected_functions_;
   absl::flat_hash_set<uint64_t> visible_functions_;
+  ThreadID selected_thread_id_ = 0;
```
Comment (florian-kuebler) [inline OrbitGl/DataManager.h]: used -1 also here.

File: OrbitGl/EventTrack.h (hunk 7 above)
Comment (florian-kuebler) [reply OrbitGl/EventTrack.h]: done.

File: OrbitGl/GpuTrack.cpp (hunk 6 above)
Comment (florian-kuebler) [reply OrbitGl/GpuTrack.cpp]: done.

File: OrbitGl/SchedulerTrack.cpp (hunk 5 above)
Comment (florian-kuebler) [reply OrbitGl/SchedulerTrack.cpp]: done.

File: OrbitGl/TextBox.cpp (hunk 4 above)
Comment (florian-kuebler) [reply OrbitGl/TextBox.cpp]: Done.

**Pull request merged**